internal/pkg/service/common/etcdop/iterator/iterator.go

```diff
@@ -49,6 +55,46 @@ func newIterator(config config) Definition {
 func (v Definition) Do(ctx context.Context, client etcd.KV, opts ...op.Option) *Iterator {
 	return &Iterator{ctx: ctx, client: client, opts: opts, config: v.config, start: v.config.prefix, page: 0, currentIndex: 0}
 }
+
+// ForEachOp method converts iterator to for each operation definition, so it can be part of a transaction.
+func (v Definition) ForEachOp(fn func(value *op.KeyValue, header *Header) error) *ForEachOp {
+	return &ForEachOp{def: v, fn: fn}
+}
+
+func (v *ForEachOp) Op(ctx context.Context) (etcd.Op, error) {
+	// If ForEachOp is combined with other operations into a transaction,
+	// then the first page is loaded within the transaction.
+	// Other pages are loaded within MapResponse method, see below.
+	// Iterator always load next pages WithRevision,
+	// so all results, from all pages, are from the same revision.
```

Comment:
Vysvetlenie ako to funguje.

Reply:
Teda pre uzivatela tohto frameworku je to transparentne, ako keby sa to dialo vsetko naraz, nad jednym snapshotom. Ale realne sa tam robi pagination, a teda nenacitava sa vsetko do memory.